### PR TITLE
Set texture api

### LINKED ISSUE #1678: Material SetURI Bug
### Description

Having two <model-viewer /> with the same `src` and trying to "setURI" in one of them changes the same material/texture on the other.

#### Live Demo
Accessing [Staging and Cameras Page](https://modelviewer.dev/examples/stagingandcameras/) and running the following code on devTools console:
`document.getElementsByTagName('model-viewer')[0].model.materials[0].pbrMetallicRoughness.baseColorTexture.texture.source.setURI('https://picsum.photos/20')`

(Works with other models as well and other URLs as well)

### Version

- model-viewer: v1.3.0

### Browser Affected

- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [ ] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [x] Windows

### PR DESCRIPTION
**Pull request by timmmeh**
### Reference Issue
Resolves issue with setURI where updating a texture on one object updates all like objects.

Fixes #1678

### CODE REVIEW

File: packages/model-viewer/src/three-components/gltf-instance/ModelViewerGLTFInstance.ts [hunk 1]
```diff
@@ -177,6 +179,24 @@ export class ModelViewerGLTFInstance extends GLTFInstance {
     }
 
     const clone = material.clone() as MeshStandardMaterial;
+    if (material.map != null) {
+      material.map = material.map.clone();
```
Comment (elalish) [inline packages/model-viewer/src/three-components/gltf-instance/ModelViewerGLTFInstance.ts:183]: This is how I tried to fix this the first time too, but it doesn't work. Cloning does not do the right thing and it ended up breaking things, see #1711 and #2279.

File: packages/model-viewer/src/features/scene-graph/material.ts [hunk 2]
```diff
@@ -115,6 +116,36 @@ export class Material extends ThreeDOMElement implements MaterialInterface {
     return (this[$sourceObject] as GLTFMaterial).emissiveFactor!;
   }
 
+  setBaseColorMap(uri: string) {
+    if (this[$pbrMetallicRoughness] !== null) {
+      this[$pbrMetallicRoughness].setBaseColorMap(uri);
+    }
+  }
+
+  setMetallicRoughnessMap(uri: string) {
```
Comment (elalish) [inline packages/model-viewer/src/features/scene-graph/material.ts:125]: I wonder if we should have these duplicated here or not. I can see both sides, but I tend toward keeping the API minimal. Do you have a strong opinion?

**Review by elalish — CHANGES_REQUESTED** [review]

File: packages/model-viewer/src/features/scene-graph/material.ts (hunk 2 above)
Comment (timmmeh) [reply packages/model-viewer/src/features/scene-graph/material.ts]: removed

File: packages/model-viewer/src/three-components/gltf-instance/ModelViewerGLTFInstance.ts (hunk 1 above)
Comment (timmmeh) [reply packages/model-viewer/src/three-components/gltf-instance/ModelViewerGLTFInstance.ts:183]: Per conversation, shininess bug is not reproducible

**Review by elalish — APPROVED** [review]
I just removed all the helper methods for now because I'd like to keep this minimal for the moment. I'm really glad you found the simple fix for this that somehow eluded me. This makes me feel much better about my understanding of three.js. Let's come back to the API shape discussion as we look to build it out further.

**Pull request merged**